Base/Python/slicer/i18n.py

```diff
@@ -51,6 +52,25 @@ def tr(text):
       statusText = _("Idle") if idle else _("Running")
 
     """
+
+    def findBracedStrings(text):
+        """Get all strings enclosed in braces, except when enclosed in double-braces."""
+        pattern = r"(?<!\{)\{([^\{\}]+)\}(?!\})"
+        matches = re.findall(pattern, text)
+        return matches
+
     filename = inspect.stack()[1][1]
     contextName = getContext(filename)
-    return translate(contextName, text)
+    translatedText = translate(contextName, text)
+
+    # Accept the translation only if all placeholders are present in the translated text to prevent runtime errors.
+    # For example:
+    #   text = "{count} items will be deleted"
+    #   translatedText = "{compter} elements seront supprimes" (incorrect, because `count` should not have been translated)
```

Comment:
```suggestion
    #   text = f"{count} items will be deleted"
    #   translatedText = f"{compter} éléments seront supprimés" (incorrect, because `count` should not have been translated)
```

Reply:
We use `.format` to make sure that no code is executed, so adding the `f` before the string could be a bit misleading.